modules/tinymce/src/core/main/ts/api/dom/ControlSelection.ts

```diff
@@ -408,38 +403,36 @@ const ControlSelection = (selection: EditorSelection, editor: Editor): ControlSe
     });
   };
 
-  const updateResizeRect = (e) => {
-    let startElm: Element, controlElm: HTMLElement;
-
-    const isChildOrEqual = (node, parent) => {
-      if (node) {
-        do {
-          if (node === parent) {
-            return true;
-          }
-        } while ((node = node.parentNode));
-      }
-    };
+  const isChildOrEqual = (node: Node | undefined, parent: Node): boolean =>
+    Type.isNonNullable(node) && dom.isChildOf(node, parent);
 
-    // Ignore all events while resizing or if the editor instance was removed
-    if (resizeStarted || editor.removed) {
+  const updateResizeRect = (e: EditorEvent<unknown>) => {
+    // Ignore all events while resizing, if the editor instance is composing or the editor was removed
+    if (resizeStarted || editor.removed || editor.composing) {
       return;
     }
 
+    const targetElm = e.type === 'mousedown' ? e.target : selection.getNode();
+    const controlElm = SelectorFind.closest<HTMLElement>(SugarElement.fromDom(targetElm), 'table,img,figure.image,hr,video,span.mce-preview-object')
```

Comment:
While you are at it maybe pull this hard coded selector out to a constant since it a pretty long line. There are a few other constants in the constructor scope unsure if it would improve the file if we moved then out to module scope.

Reply:
Done everything that I thought made sense (I left the `rootDocument` as it was clearer beside `rootElement`).